### DISCUSSION TITLE
Web protocol is missing from the link of the devices in a rack

### DISCUSSION THREAD

**Discussion by qmgwb** [netbox-community/netbox#6209] [Q&A] [answered]
### NetBox version

v2.11

### Python version

3.6

### Steps to Reproduce

1. Open Rack, i.e. /dcim/racks/1/
2. Hover over the link of a device within the rack view
3. Verify the link only shows the host + path and resource, but not the web protocol, i.e. https
4. Clicking the link will result in "not found"

### Expected Behavior

Opening the device page using https. 

### Observed Behavior

Opening the http server page - showing a 404.

**Comment by jeremystretch** [comment, marked as answer]
This occurs when the HTTP frontend is not configured to forward the protocol correctly. Please see the provided example [nginx](https://github.com/netbox-community/netbox/blob/develop/contrib/nginx.conf) and [Apache](https://github.com/netbox-community/netbox/blob/develop/contrib/apache.conf) configurations for reference.

**Reply by qmgwb** [reply]
Thanks and you are right. Still - all other links within Netbox provide the correct protocol prefix (https://), so I was wondering if that could be added here as well.

**Reply by candlerb** [reply]
Netbox doesn't know whether the proxy in front of it is serving HTTP or HTTPS.  The links you see with the "correct protocol prefix" are almost certainly relative links - e.g. `/dcim/devices/1` rather than `https://yourdomain.com/dcim/devices/1` - and it's the browser which completes them.  You can view the raw HTML to confirm this.

When Netbox does need to generate an absolute link, it relies on the proxy setting headers such as `X-Forwarded-Proto` and `X-Forwarded-Host` to construct the correct link.
